graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java

```diff
@@ -96,6 +98,11 @@ public ScriptingApiResource(final SearchExecutor searchExecutor,
     @NoAuditEvent("Creating audit event manually in method body.")
     public TabularResponse executeQuery(@ApiParam(name = "queryRequestSpec") @Valid MessagesRequestSpec messagesRequestSpec,
                                         @Context SearchUser searchUser) {
+
+        if (messagesRequestSpec.from() + messagesRequestSpec.size() > MAX_RESULT_WINDOW) {
```

Comment:
Wouldn't this also throw an error and abort searches for users which have raised the maximum result window?

Reply:
Yes, it would. 
The question is if we want to take care of this unusual/unlikely possibility and perform additional query to check this setting for an index we query (adding some performance overhead) or are we ok with the current, simple approach? It is hard for me to imagine people getting thousands of messages with scripting API, so I am a little bit hesitant to add complexity and perf. overhead just to make it extremely bulletproof...

Reply:
What would be the benefit of the current approach? Either way the user would get back an error/exception when the overall message size exceeds `n` (defaulting to 10000). If we are worried about the default message coming back from ES/OS is not understandable, wouldn't it be better to just catch it and return a better one instead of doing a pre-flight check?

Reply:
The benefit is we don't reach ES/OS at all in this case.
But if you believe it is not worth it, or risky, we can throw this PR away, it is nothing big anyway...